github/repos_rules.go

```diff
@@ -119,14 +119,18 @@ func (r *RepositoryRule) UnmarshalJSON(data []byte) error {
 		r.Parameters = nil
 	case "update":
 		params := UpdateAllowsFetchAndMergeRuleParameters{}
-		if err := json.Unmarshal(*RepositoryRule.Parameters, &params); err != nil {
-			return err
-		}
+		if RepositoryRule.Parameters != nil {
```

Comment:
Let's please change this to make it easier to read:

```go
if RepositoryRule.Parameters == nil {
  r.Parameters = nil
  return nil
}
```

Reply:
Right, I understand, but I want to highlight the "happy path" and provide the early return which is idiomatic Go.
Please switch to my suggestion, and feel free to put a blank line after the new block if you think it reads better.

Reply:
Done, also added a bit in the update rule constructor to check for empty params since this specific rule is a bit different than others 🙂